core/src/mender-client.c

```diff
@@ -244,13 +251,17 @@ mender_err_t
 mender_client_init(mender_client_config_t *config, mender_client_callbacks_t *callbacks) {
 
     assert(NULL != config);
-    assert(NULL != config->device_type);
     assert(NULL != callbacks);
     assert(NULL != callbacks->restart);
     mender_err_t ret;
 
-    mender_client_config.device_type = config->device_type;
-
+    if ((NULL != config->device_type) && (0 != strlen(config->device_type))) {
+        /* Use client config unless string is NULL or empty */
+        mender_client_config.device_type = config->device_type;
+    } else {
+        /* Otherwise, use Kconfig */
+        mender_client_config.device_type = CONFIG_MENDER_DEVICE_TYPE;
+    }
```

Comment:
`CONFIG_MENDER_DEVICE_TYPE` can be null, so:
```suggestion
    } else if  (NULL != CONFIG_MENDER_DEVICE_TYPE ){
        /* Otherwise, use Kconfig */
        mender_client_config.device_type = CONFIG_MENDER_DEVICE_TYPE;
    } else {
       /* Error... */
    }
```

Reply:
Or to have an explicit follow-up check, as it is done for `mender_client_config.host` :point_down: